crates/fluvio/src/producer/mod.rs

```diff
@@ -90,24 +94,72 @@ impl ProducerPool {
         }
     }
 
-    fn shared(
+    #[allow(clippy::too_many_arguments)]
+    pub async fn add_producer_if_needed(
+        &mut self,
         config: Arc<TopicProducerConfig>,
         topic: String,
         spu_pool: Arc<SpuPool>,
-        batches: Arc<Vec<BatchHandler>>,
         client_metric: Arc<ClientMetrics>,
-    ) -> Arc<Self> {
-        Arc::new(ProducerPool::new(
+        partition_id: PartitionId,
+        batchs: Arc<BatchesDeque>,
+        batch_events: Arc<BatchEvents>,
+        record_accumulator: Arc<RecordAccumulator>,
```

Comment:
having too many arguments may indicate need to put them in a struct?

Reply:
Fixed!